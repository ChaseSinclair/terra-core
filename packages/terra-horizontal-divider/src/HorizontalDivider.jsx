import React from 'react';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import styles from './HorizontalDivider.scss';

const cx = classNames.bind(styles);

const HorizontalDivider = ({ ...customProps }) => {
  const HorizontalDividerClassNames = cx([
    'divider',
    customProps.className,
  ]);

  return (<hr {...customProps} className={HorizontalDividerClassNames} />);
};

export default HorizontalDivider;
