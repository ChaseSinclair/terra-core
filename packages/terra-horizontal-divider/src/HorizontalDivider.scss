:local {
  .divider {
    border-color: var(--terra-horizontal-divider-border-color, #b2b5b6);
    border-style: solid;
    border-top-width: var(--terra-horizontal-divider-border-width, 1px);
    height: 0;
    margin: 0;
  }
}
