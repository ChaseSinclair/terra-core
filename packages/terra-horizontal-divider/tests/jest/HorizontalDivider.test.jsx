import React from 'react';
import HorizontalDivider from '../../src/HorizontalDivider';

describe('HorizontalDivider', () => {
  const defaultRender = <HorizontalDivider />;

  // Snapshot Tests
  it('should render a basic component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  // Structure Tests
  it('should have the class divider', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.prop('className')).toContain('divider');
  });

  // Custom Prop Test
  it('it should pass in a custom prop', () => {
    const wrapper = shallow(<HorizontalDivider id="testDivider" />);
    expect(wrapper.unrendered.props.id).toEqual('testDivider');
  });
});
