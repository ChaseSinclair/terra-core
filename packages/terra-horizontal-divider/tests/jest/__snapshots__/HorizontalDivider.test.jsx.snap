// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalDivider should render a basic component 1`] = `
<hr
  className="divider"
/>
`;
